crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -355,6 +360,76 @@ impl Timeline {
         }
     }
 
+    /// Send a reply to the given event.
+    ///
+    /// Currently only supports events events with an event ID, and will log an
+    /// error and do nothing for other events. Please check
+    /// [`EventTimelineItem::can_be_replied_to`] before calling this.
+    ///
+    /// # Arguments
+    ///
+    /// * `content` - The content of the reply
+    ///
+    /// * `reply_item` - The event item you want to reply to
+    ///
+    /// * `forward_thread` - Usually `Yes`, unless you explicitly want to the
+    ///   reply to show up in the main timeline even though the `reply_item` is
+    ///   part of a thread
+    ///
+    /// * `add_mentions` - Set to `Yes` if the `mentions` of `content` are
+    ///   propagated according to user intent, `No` otherwise
+    ///
+    /// * `txn_id` - Optional transaction ID, usually `None`
+    #[instrument(skip(self, content, reply_item, txn_id))]
+    pub async fn send_reply(
+        &self,
+        content: RoomMessageEventContent,
+        reply_item: &EventTimelineItem,
+        forward_thread: ForwardThread,
+        add_mentions: AddMentions,
+        txn_id: Option<&TransactionId>,
+    ) {
+        // Early returns here must be in sync with
+        // `EventTimelineItem::can_be_replied_to`
+        let Some(event_id) = reply_item.event_id() else {
+            error!("send_reply does not currently support replying to local echoes");
```

Comment:
I kind of get what you mean, but AFAIK EXI just logs any errors and does nothing more itself. By doing the logging here the log message has more details, and we don't have to extend the error enum or introduce a new error type. I'm okay with changing it for other consumers I guess.. Would you prefer a separate error type (I *think* I do personally, because this is the kind of thing that practically never makes sense to bubble up) or an addition to the timeline's `Error` type?